src/build.ts

```diff
@@ -137,9 +137,19 @@ async function _build (nitro: Nitro, rollupConfig: RollupConfig) {
   await _snapshot(nitro)
 
   nitro.logger.start('Building server...')
-  const build = await rollup.rollup(rollupConfig).catch((error) => {
-    nitro.logger.error('Rollup error: ' + error.message)
-    throw error
+  const build = await rollup.rollup(rollupConfig).catch((_error) => {
+    try {
+      for (const error of ('errors' in _error ? _error.errors : [_error])) {
+        const id = error.id || _error.id
+        let path = isAbsolute(id) ? relative(process.cwd(), id) : id
+        const location = error.loc || error.location
+        if (location) {
+          path += `:${location.line}:${location.column}`
+        }
+        nitro.logger.error(`Rollup error while processing \`${path}\`` + '\n' + '\n' + (error.text || error.frame))
+      }
+    } catch {}
```

Comment:
Why another error should happen? If it is our internal code, it shouldn't be hidden like this IMO.